File: solutions/0035. Search Insert Position/0035.py
# Time complexity: O(logn)
# Space complexity: O(1)
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        start = 0
        end = len(nums)
       

        while start < end:
            mid = start + (end - start) // 2
            if nums[mid] == target:
                return mid
            if nums[mid] > target:
                end = mid 
            if nums[mid] < target:
                start = mid + 1

        return start

File: solutions/0035. Search Insert Position/0035.js
// Time complexity: O(logn)
// Space complexity: O(1)
var searchInsert = function(nums, target) {
  let left = 0;
  let right = nums.length;

  while (left  < right) {
    let mid = left + Math.floor((right - left) / 2);
        
    if (nums[mid] == target)
      return mid;
    
    if (nums[mid] < target)
      left = mid + 1;
    else
      right = mid;
  }
  
  return left;
};